pkg/extension/wasmimage/wasmimage.go

```diff
@@ -85,3 +88,25 @@ func isWasmBinary(path string) bool {
 	}
 	return bytes.Equal(buffer, magic)
 }
+
+func registryHosts(insecure, plainHTTP bool) docker.RegistryHosts {
+	client := http.DefaultClient
+
+	if insecure && !plainHTTP {
+		client.Transport = &http.Transport{TLSClientConfig: &tls.Config{InsecureSkipVerify: true}} // nolint:gosec
+	}
+
+	return func(host string) ([]docker.RegistryHost, error) {
+		config := docker.RegistryHost{
+			Client:       client,
+			Host:         host,
+			Scheme:       "https",
+			Path:         "/v2", // For Docker registry:2 https://docs.docker.com/registry/spec/api/#overview
```

Comment:
what if we push against other registries..? This should work with ECR and GCR as well (but unfortunately we don't have test infra right now). Maybe we could just remove `/v2`..?

Reply:
anyway I have to refactor the logic this packaging stuff since the OCI spec has changed since the PR for this feature was merged.. so we could leave this as is now.

Reply:
this isn't a change btw, before you just didn't see it adding /v2

Reply:
for example, the mock test I recently added? it had /v2 in there because the old client sent it also. If we delete it, not only will it not work, but the mock test will show it won't.

Reply:
ps most of this is following down the rabbit hole https://github.com/deislabs/oras/blob/a64b0ac4faf26c58016fbd7ed1a9eb9d049e54ee/cmd/oras/resolver.go because we support old flags you would end up noticing the default from docker is hard-coding the path to /v2

Reply:
I see, that makes sense! Thanks for clarifying! 💯